src/main/java/pl/javastart/bootcamp/domain/admin/trainer/AdminTrainerController.java

```diff
@@ -0,0 +1,65 @@
+package pl.javastart.bootcamp.domain.admin.trainer;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.List;
+
+@Controller
+public class AdminTrainerController {
+    private final AdminTrainerService adminTrainerService;
+
+    public AdminTrainerController(AdminTrainerService adminTrainerService) {
+        this.adminTrainerService = adminTrainerService;
+    }
+
+    @GetMapping("/admin/trenerzy")
+    public String trainerList(Model model) {
+        List<Trainer> trainers = adminTrainerService.findAll();
+        model.addAttribute("trainers", trainers);
+        return "admin/trainer/trainerList";
+    }
+
+    @GetMapping("/admin/trenerzy/{id}")
+    public String viewTrainer(@PathVariable Long id, Model model) {
+        Trainer trainer = adminTrainerService.findByIdOrThrow(id);
+        model.addAttribute("trainer", trainer);
+        return "admin/trainer/trainer";
+    }
+
+    @PostMapping("/admin/trenerzy/{id}/edytuj")
+    public String changeTrainerData(@PathVariable Long id, Model model) {
+        Trainer trainer = adminTrainerService.findByIdOrThrow(id);
+        model.addAttribute("trainer", trainer);
+        model.addAttribute("mode", "edit");
+        return "admin/trainer/trainerAddOrEdit";
+    }
```

Comment:
To pozostaw jako GET. Dopiero żądanie wysłane z formularza edycji, czyil wywołanie metody editTrainer powinno być przez POST (co już masz).
Analogicznie jeżeli miałbyś np. dwuetapowe usuwanie, czyli najpierw wyświetlenie strony z potwierdzeniem "czy na pewno chcesz usunąć ABC?", a dopiero tam przycisk z potwierdzeniem, to wtedy też byłyby dwie metody, 
najpierw GET pod /admin/trenerzy/usun/123 (potwierdzenie)
później POST pod /admin/trenerzy/usun/123 (faktyczne usuwanie)